src/ansys/dpf/post/simulation.py

```diff
@@ -659,26 +659,22 @@ def _build_selection(
         if requires_manual_averaging and location != locations.elemental_nodal:
             location = locations.elemental_nodal
 
+        result_native_location = None
+        if base_name in _result_properties:
+            result_native_location = _result_properties[base_name].get("location")
```

Comment:
I'd prefer to avoid using the `_result_properties` dict if possible (https://github.com/ansys/pydpf-core/issues/2366). I know that it is currently being used in other places here, but this needs to be refactored. Could you work instead with the ResultInfo? Thanks!